### PR TITLE
[ci] Add Apple M1 buildbot

### PR DESCRIPTION
**Pull request by ljcc0930**
Related issue = #2254, #2255

Unpassed tests:
test_custom_type_atomics.py::test_custom_int_atomics
test_bit_struct.py::test_custom_int_load_and_store
test_struct_for_intermediate.py::test_nested

Randomly crashed:
test_gc.py::test_block_async
test_sfg.py::test_remove_clear_list_from_fused_serial

Banned quant and async on metal

### CODE REVIEW

File: tests/python/test_bit_struct.py [hunk 1]
```diff
@@ -36,7 +36,7 @@ def verify_val():
     verify_val.__wrapped__()
 
 
-@ti.test(require=ti.extension.quant_basic, debug=True)
+@ti.test(require=ti.extension.quant_basic, exclude=[ti.metal], debug=True)
```
Comment (ailzhang) [inline tests/python/test_bit_struct.py:39]: Nit: if we exclude a backend, it'd be nice if we can add a note whether we need to fix this later (like if it's supposed to be excluded) ;)

**Review by ailzhang — APPROVED** [review]
Looks great! Thanks!

File: tests/python/test_bit_struct.py (hunk 1 above)
Comment (ljcc0930) [reply tests/python/test_bit_struct.py:39]: I'll start a new issue

File: .github/workflows/presubmit.yml [hunk 2]
```diff
@@ -276,3 +276,46 @@ jobs:
...
+          arch -arm64 python3 -m pip install dist/*.whl
+        env:
+          CI_SETUP_CMAKE_ARGS: -DTI_WITH_OPENGL:BOOL=OFF -DTI_WITH_CUDA:BOOL=OFF -DTI_WITH_CC:BOOL=${{ matrix.with_cc }} -DTI_BUILD_TESTS:BOOL=${{ matrix.with_cpp_tests }}
+
+      - name: Test
+        run: |
+          export PATH=$PATH:$HOME/Library/Python/3.8/bin
+          arch -arm64 python3 examples/algorithm/laplace.py
+          arch -arm64 ti diagnose
+          arch -arm64 ti changelog
```
Comment (k-ye) [inline .github/workflows/presubmit.yml:316]: Remove

File: .github/workflows/presubmit.yml [hunk 3]
```diff
@@ -276,3 +276,46 @@ jobs:
...
+
+  build_and_test_m1:
+    name: Build and Test (Apple M1)
+    needs: check_code_format
+    strategy:
+      matrix:
+        include:
+          - os: macos-latest
+            python: 3.8
+            with_cc: OFF
```
Comment (k-ye) [inline .github/workflows/presubmit.yml:288]: Let's just turn off this as well as always enable CPP tests

File: tests/python/test_runtime.py [hunk 4]
```diff
@@ -60,7 +60,7 @@ def patch_os_environ_helper(custom_environ: dict, excludes: dict):
     'flatten_if': [False, TF],
     'simplify_before_lower_access': [True, TF],
     'simplify_after_lower_access': [True, TF],
-    'use_unified_memory': [ti.get_os_name() != 'win', TF],
+    'use_unified_memory': [ti.get_os_name() not in ['win', 'osx'], TF],
```
Comment (k-ye) [inline tests/python/test_runtime.py:63]: Just use 'linux'?

File: .github/workflows/presubmit.yml [hunk 5]
```diff
@@ -276,3 +276,46 @@ jobs:
...
+        run: |
+          arch -arm64 python3 -m pip uninstall taichi -y
+          git --version
+          export CXX=clang++
+          arch -arm64 python3 -m pip install -r requirements_dev.txt
+          cd python
+          git fetch origin master
+          TAICHI_CMAKE_ARGS=$CI_SETUP_CMAKE_ARGS arch -arm64 python3 build.py build
+          cd ..
+          echo `ls dist/*.whl`
```
Comment (Leonz5288) [inline .github/workflows/presubmit.yml:306]: I believe this line is unnecessary?

**Review by k-ye — APPROVED** [review]
LGTM thanks!

**Pull request merged**